torch_np/_detail/implementations.py

```diff
@@ -118,3 +368,101 @@ def meshgrid(*xi_tensors, copy=True, sparse=False, indexing="xy"):
         output = [x.clone() for x in output]
 
     return output
+
+
+def bincount(x_tensor, /, weights_tensor=None, minlength=0):
+    int_dtype = _dtypes_impl.default_int_dtype
+    (x_tensor,) = _util.cast_dont_broadcast((x_tensor,), int_dtype, casting="safe")
+
+    result = torch.bincount(x_tensor, weights_tensor, minlength)
+    return result
+
+
+# ### linspace, geomspace, logspace and arange ###
+
+
+def geomspace(start, stop, num=50, endpoint=True, dtype=None, axis=0):
+    if axis != 0 or not endpoint:
+        raise NotImplementedError
+    base = torch.pow(stop / start, 1.0 / (num - 1))
+    logbase = torch.log(base)
+    result = torch.logspace(
+        torch.log(start) / logbase,
+        torch.log(stop) / logbase,
+        num,
+        base=base,
+    )
+    return result
+
+
+def arange(start=None, stop=None, step=1, dtype=None):
+    if step == 0:
+        raise ZeroDivisionError
+    if stop is None and start is None:
+        raise TypeError
+    if stop is None:
+        # XXX: this breaks if start is passed as a kwarg:
+        # arange(start=4) should raise (no stop) but doesn't
+        start, stop = 0, start
+    if start is None:
+        start = 0
+
+    if dtype is None:
+        dt_list = [_util._coerce_to_tensor(x).dtype for x in (start, stop, step)]
+        dtype = _dtypes_impl.default_int_dtype
+        dt_list.append(dtype)
+        dtype = _dtypes_impl.result_type_impl(dt_list)
+
+    try:
+        return torch.arange(start, stop, step, dtype=dtype)
+    except RuntimeError:
+        raise ValueError("Maximum allowed size exceeded")
+
+
+# ### empty/full et al ###
+
+
+def eye(N, M=None, k=0, dtype=float):
+    if M is None:
+        M = N
+    z = torch.zeros(N, M, dtype=dtype)
+    z.diagonal(k).fill_(1)
+    return z
+
+
+def zeros_like(a, dtype=None, shape=None):
+    result = torch.zeros_like(a, dtype=dtype)
+    if shape is not None:
+        result = result.reshape(shape)
```

Comment:
Fantastic candidate to be implemented via a decorator

Reply:
Let's keep this in mind for future refactors.